server/test/repository/ApplicationStatusesRepositoryTest.java

```diff
@@ -65,11 +67,10 @@ public void lookupListOfObsoleteStatusDefinitions_throwsException() {
   public void canQueryForAllApplicationStatuses() {
     Long uniqueProgramId = new Random().nextLong();
     ProgramModel program =
-        ProgramBuilder.newActiveProgram("test program" + uniqueProgramId, "description")
-            .withStatusDefinitions(new StatusDefinitions(ImmutableList.of(REAPPLY_STATUS)))
-            .build();
-
+        ProgramBuilder.newActiveProgram("test program" + uniqueProgramId, "description").build();
     String programName = program.getProgramDefinition().adminName();
+    repo.createOrUpdateStatusDefinitions(
+        programName, new StatusDefinitions(ImmutableList.of(REAPPLY_STATUS)));
 
     StatusDefinitions statusDefinitions = new StatusDefinitions(ImmutableList.of(APPROVED_STATUS));
```

Comment:
nitpick (non-blocking): Would it work to remove some lines by adding more than one status at a time? Eg. 
```
repo.createOrUpdateStatusDefinitions(
        programName, new StatusDefinitions(ImmutableList.of(REAPPLY_STATUS, APPROVED_STATUS)));
```

Reply:
no actually the tests is query for all histrorical + active status so that we can use the status text for filtering applications. Technically it doesn't matter if the query returns 3 or 4. We should see both Reapply and Approved status text in the final result. But I am also surprised that the program.build() is getting called twice

Reply:
fixed it. Thanks @dkatzz . Adding it after the block.build() worked!